apps/s_client.c

```diff
@@ -3467,4 +3509,25 @@ static int ldap_ExtendedResponse_parse(const char *buf, long rem)
     return ret;
 }
 
+static char *base64encode (const void *buf, size_t len) {
+
+    /* BASE64 encoder - used only for encoding basic proxy authentication credentials
+     * Based on an idea by schulwitz here: 
+     * https://stackoverflow.com/questions/342409/how-do-i-base64-encode-decode-in-c/
```

Comment:
You cannot copy & paste code from stackoverflow into OpenSSL since you do not own the copyright on it and you would be relicensing it. We require all contributors to have signed a CLA, and this is not true for the original author in this case. In any case I suggest you consider rewriting this function to use the EVP_Encode* functions instead:

https://www.openssl.org/docs/manmaster/man3/EVP_EncodeInit.html

Reply:
Thanks. Seems EVP_EncodeBlock() is a better fit, as it doesn't add a newline after every 64 chars.

I've reworked this function now.